targets/mega65/uart_monitor.c

```diff
@@ -108,9 +110,22 @@ static void m65mon_dumpmem28 ( int addr )
 {
 	int n = 16;
 	addr &= 0xFFFFFFF;
-	umon_printf(":%07X:", addr);
+	umon_printf(":%08X:", addr);
 	while (n--)
-		umon_printf("%02X", debug_read_linear_byte(addr++));
+	{
```

Comment:
Why I chose to implement it this way over what you had previously is because if you see how the hardware does it, when you type something like: "M 777d000" you see in the output an address of "0777xxxx":
```
<dbg>M 777d000
:0777D000:00000000000000000000000000000000
:0777D010:001BA8000000C9002460E10000000000
```

...but in xemu presently, it would output with an address of "000xxxx":
```
<dbg>M 777d000
:000D000:00000000000000000000000000000000
:000D010:001B00000000C9002470F10000000000
```